pkg/logql/downstream_test.go

```diff
@@ -334,8 +340,14 @@ func TestApproxTopkSketches(t *testing.T) {
 			regularQuery:  `topk(100, sum by (a) (sum_over_time ({a=~".+"} | logfmt | unwrap value [1s])))`,
 			realtiveError: 0.0015,
 		},
+		{
+			labelShards:   10, // increasing this will make the test too slow
+			totalStreams:  1_000_000,
+			shardedQuery:  `approx_topk(100, sum by (a) (sum_over_time ({a=~".+"} | logfmt | unwrap value [1s])))`,
+			regularQuery:  `topk(100, sum by (a) (sum_over_time ({a=~".+"} | logfmt | unwrap value [1s])))`,
+			realtiveError: 0.0015,
+		},
```

Comment:
I think this is one of the things that is making some these test results seem better than they would be in reality; here you've got `labelShards: 10`, which means that there will only be 10 unique values for the label `a` but we're doing `approx_topk(100, ...)` as the query.

Reply:
You are right. So this test is actually no good. What we need are 1 million unique values...